File: AILang/benchmarks/python/recursive_traversal.py
def recursive_fib(n: int) -> int:
    if n <= 1:
        return 1
    return recursive_fib(n - 1) + recursive_fib(n - 2)


def recursive_bench(iterations: int) -> int:
    return recursive_fib(iterations)


def main() -> int:
    depth = 32
    result = recursive_bench(depth)
    print(result)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: AILang/benchmarks/rust/recursive_traversal.rs
fn recursive_fib(n: i32) -> i64 {
    if n <= 1 {
        return 1;
    }
    recursive_fib(n - 1) + recursive_fib(n - 2)
}

fn recursive_bench(iterations: i32) -> i64 {
    recursive_fib(iterations)
}

fn main() {
    let depth = 32;
    let result = recursive_bench(depth);
    println!("{}", result);
}
